recipes/libpqxx/all/conanfile.py

```diff
@@ -0,0 +1,94 @@
+    homepage = "https://github.com/jtv/libpqxx"
+    license = "BSD-3-Clause"
+    topics = ("conan", "libpqxx", "postgres", "postgresql", "database", "db")
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    requires = "libpq/11.5"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        compiler_version = Version(self.settings.compiler.version.value)
+
+        minimal_version = {
+            "Visual Studio": "15",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10"
+        }
+
+        if compiler in minimal_version and \
+           compiler_version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports"
+                                            " at least C++17. %s %s is not"
+                                            " supported." % (self.name, compiler, compiler_version))
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "17")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["BUILD_DOC"] = False
+            self._cmake.definitions["BUILD_TEST"] = False
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_files(self):
+        if self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+
+    def build(self):
+        self._patch_files()
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["pqxx"]
+
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs = ["wsock32 ", "ws2_32"]
+        elif self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["pthread"]
```

Comment:
Are you sure about pthread? 

```
ldd lib/libpqxx-7.0.so
	linux-vdso.so.1 (0x00007fffababe000)
	libpq.so.5 => not found
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f7a487d9000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f7a48693000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f7a48679000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f7a484b3000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f7a48a76000)
```

Also, I didn't find any relation on source_folder:

```
grep pthread source_subfolder/ -r
source_subfolder/config/ltmain.sh:	   # Do not include libc_r directly, use -pthread flag.
source_subfolder/config/ltmain.sh:      -mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe \
source_subfolder/config/ltmain.sh:	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe \
```

Reply:
Probably an old dependency.
I' ll remove it.

Reply:
@uilianries Do you do this check manually or is there a hook for it?

Reply:
@madebr manually, but we have the idea of moving it to a hook

Reply:
Removed pthread: be697bc9b5037f606156e95eb29070bf57f24a26